uwblueprint/headsupguys
# Tool Builder - Self Check Responses (FRONT-END)

## Notion ticket link

[[Tool Builder] Self Check Response](https://www.notion.so/uwblueprintexecs/Task-Board-485f846ae00243599dddba2be9bcba82?p=eefac0f1739049ce92abbf45673b9f6b)



## Implementation description
* Used array of breakpoints
* Each breakpoint has a component attached to it
* The component is created using the data from the breakpoint data in the array
* Creates the new breakpoint by setting the lower as the last breakpoint's upper + 1



## Steps to test
1. Open http://localhost:3000/admin/dashboard/tools
2. Create or edit an existing tool
3. Go to Self Check tab
4. Click Response
5. Explore! You can try to add new breakpoints, change the screen size, set the min and max of the breakpoints, add questions, etc. Note that you will have to create questions for the self check that are either multiple choice or multiselect to be able to play around with this functionality.

NOTE: the back-end will be done in another PR.



## What should reviewers focus on?
* Error checking
* Functions to compute the selected questions titles... Should I use an array with the useState hook to keep track of the number for the questions that are added? ~~Would have to sort these though~~ Just realized that I still have to sort them (or add them in sorted order)..
* Any other suggestions to make the code better! :)
* Let me know if there's something I forgot with this ticket!


## Checklist
